code/controllers/subsystem/lighting.dm

```diff
@@ -0,0 +1,87 @@
+GLOBAL_LIST_EMPTY(lighting_update_lights) // List of lighting sources  queued for update.
+GLOBAL_LIST_EMPTY(lighting_update_corners) // List of lighting corners  queued for update.
+GLOBAL_LIST_EMPTY(lighting_update_objects) // List of lighting objects queued for update.
+
+SUBSYSTEM_DEF(lighting)
+	name = "Lighting"
+	wait = 2
+	init_order = INIT_ORDER_LIGHTING
+	flags = SS_TICKER
+
+/datum/controller/subsystem/lighting/stat_entry()
+	..("L:[GLOB.lighting_update_lights.len]|C:[GLOB.lighting_update_corners.len]|O:[GLOB.lighting_update_objects.len]")
+
+/datum/controller/subsystem/lighting/Initialize(timeofday)
+	if(!initialized)
+		if(config.starlight)
+			for(var/I in GLOB.all_areas)
+				var/area/A = I
+				if (A.dynamic_lighting == DYNAMIC_LIGHTING_IFSTARLIGHT)
+					A.luminosity = 0
+
+		create_all_lighting_objects()
+		initialized = TRUE
+
+	fire(FALSE, TRUE)
+
+	return ..()
+
+/datum/controller/subsystem/lighting/fire(resumed, init_tick_checks)
+	MC_SPLIT_TICK_INIT(3)
+	if(!init_tick_checks)
+		MC_SPLIT_TICK
+	var/i = 0
+	for(i in 1 to GLOB.lighting_update_lights.len)
+		var/datum/light_source/L = GLOB.lighting_update_lights[i]
+
+		L.update_corners()
+
+		L.needs_update = LIGHTING_NO_UPDATE
+
+		if(init_tick_checks)
+			CHECK_TICK
+		else if (MC_TICK_CHECK)
```

Comment:
okay personal nitpick but i always hated the space between the if and the paren thing.

Reply:
Our style guide agrees haha. No spesses

Reply:
I'll have to adjust this. /tg/ doesn't really care about this, which is why it's inconsistent. 

Honestly, I think we shouldn't worry so much about these small issues, like /tg/ does. We don't have anywhere near as many coders or reviewers as them, yet we tend to have much more in-depth reviews and care much more about issues like these. Personally, I think we'd be much better off loosening up in that regard so we can get more (new) stuff in.